### DISCUSSION TITLE
Welcome to Microsoft Blazor Book 3 Discussions!

### DISCUSSION THREAD

**Discussion by PeterHimschoot** [PeterHimschoot/microsoft-blazor-book-3#1] [Announcements]
## 👋 Welcome!
  Please put your questions and remarks here.
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.

**Comment by SurajKapadia** [comment]
I created a Server-Side application, and ran into an issue.  So, I thought to post the solution here if someone is having same issue.

Exception - 
System.InvalidOperationException: The current thread is not associated with the Dispatcher.

Page 68 - Listing 2-25.

The example works for Client-Side application.
For the Server-Side application, need to enclose in InvokeAsync method.
callback: (_) => { IncrementCount(); InvokeAsync(StateHasChanged); },

I found the solution here - 
https://blazor-university.com/components/multi-threaded-rendering/invokeasync/

Thanks,
Suraj Kapadia.
